### PR TITLE
Limit servicing staff users to API GETs

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: temba/api/models.py [hunk 1]
```diff
@@ -74,21 +74,23 @@ def has_permission(self, request, view):
...
+            if role not in APIToken.ALLOWED_ROLES:
                 return False
         elif org:
             role = org.get_user_role(request.user)
         else:
             return False
 
         has_perm = request.user.is_staff or role.has_api_perm(permission)
+        if not has_perm:
+            return False
```
Comment (norkans7) [inline temba/api/models.py:87]: staff will always return False?

Comment (norkans7) [reply temba/api/models.py:87]: never mind

File: temba/api/models.py [hunk 2]
```diff
@@ -74,21 +74,23 @@ def has_permission(self, request, view):
...
 
-        # viewers can only ever GET from the API
-        if role == OrgRole.VIEWER:
-            return has_perm and request.method == "GET"
+        # viewers and servicing staff can only ever GET from the API
+        if role == OrgRole.VIEWER or (not role and request.user.is_staff):
+            return request.method == "GET"
```
Comment (norkans7) [inline temba/api/models.py:91]: I think for viewers we want to keep `has_perm and request.method == "GET"` so viewers cannot list things like users as the are not allowed to access those lists

Comment (rowanseymour) [reply temba/api/models.py:91]: They specifically have that as an API only permission because the editor requires it

Comment (norkans7) [reply temba/api/models.py:91]: I see line 86-87 will have prevented to reach here if the permissions is not allowed

**Review by norkans7 — APPROVED** [review]

**Pull request merged**